docs/design/2026_04_28_proposed_keyviz_adapter_labels.md

```
@@ -0,0 +1,785 @@
+`dynamo/users` the composite becomes `route:1:dynamo/users` and
+parsers split on the *first* `:` after `route`.
+
+**Hard constraint on canonical labels**: a label MUST NOT contain
+`:`. The constants in `keyviz/labels.go` (§9 Q2) are the only
+source of label values, and we enforce this at the constants
+file with a `func TestAllLabelsAvoidSeparator(t *testing.T)` test
+that asserts `!strings.ContainsRune(string(l), ':')` for every member of
+`AllLabels`. Without this constraint a future label like
+`"redis:db0"` would silently break `bucket_id` parsing — the
+parser would split at the wrong `:`. (CodeRabbit major on PR
+#694.)
```

markdownlint-cli2 (0.22.1)

[warning] No space after hash on atx style heading

(MD018, no-missing-space-atx)